src/gui/rss/automatedrssdownloader.cpp

```diff
@@ -315,9 +311,6 @@ void AutomatedRssDownloader::updateRuleDefinitionBox(QListWidgetItem *selected)
             else
                 lMatch = tr("Last Match: Unknown");
             ui->lblLastMatch->setText(lMatch);
-            updateMustLineValidity();
-            updateMustNotLineValidity();
-            updateEpisodeFilterValidity();
```

Comment:
These lines are necessary for ensuring the text fields are re-checked for validity when a new rule is selected. Otherwise if the old selection had an invalid entry these won't be updated correctly.